src/sentry/tasks/relay.py

```diff
@@ -32,6 +32,23 @@ def update_config_cache(
     from sentry.relay import projectconfig_cache
     from sentry.relay.config import get_project_config
 
+    # At this point, we haven't marked the task as completed, so it should be
+    # debounced. If it isn't, another worker has picked a task for the same
+    # public key and marked it as done. Since we want to avoid recomputation,
+    # exit the current task.
+    if not projectconfig_debounce_cache.is_debounced(
+        public_key=public_key, project_id=None, organization_id=None
+    ):
+        return
```

Comment:
Haven't thought this through, but do we really want to change this logic? Until now the debouncer was just for scheduling, not running.

Reply:
I think it prevented something, but I can't remember what that is after a few minutes of thinking. I'll remove it.

Reply:
I can't think of any issues, except the theoretical case that the debounce flag expires and silently disappears from redis. So I guess it's ok to keep this, but I would go one step further and make the "check and set" atomic, i.e. make `mark_task_done` return a boolean that is `True` if the flag was successfully deleted (I think @jan-auer  suggested using the [return value of redis' `DEL` operation](https://redis.io/commands/del/#return) for this).